public/app/plugins/panel/bargauge/suggestions.ts

```diff
@@ -1,115 +1,60 @@
-    if (dataSummary.hasStringField && dataSummary.frameCount === 1 && dataSummary.rowCountTotal < 30) {
-      list.append({
-        name: SuggestionName.BarGaugeBasic,
-        options: {
-          reduceOptions: {
-            values: true,
-            calcs: [],
-          },
-          displayMode: BarGaugeDisplayMode.Basic,
-          orientation: VizOrientation.Horizontal,
-        },
-        fieldConfig: {
-          defaults: {
-            color: {
-              mode: FieldColorModeId.ContinuousGrYlRd,
-            },
-          },
-          overrides: [],
-        },
-      });
+const BAR_LIMIT = 30;
 
-      list.append({
-        name: SuggestionName.BarGaugeLCD,
-        options: {
-          reduceOptions: {
-            values: true,
-            calcs: [],
-          },
-          displayMode: BarGaugeDisplayMode.Lcd,
-          orientation: VizOrientation.Horizontal,
-        },
-        fieldConfig: {
-          defaults: {
-            color: {
-              mode: FieldColorModeId.ContinuousGrYlRd,
-            },
-          },
-          overrides: [],
-        },
-      });
-    } else {
-      list.append({
-        name: SuggestionName.BarGaugeBasic,
-        options: {
-          displayMode: BarGaugeDisplayMode.Basic,
-          orientation: VizOrientation.Horizontal,
-          reduceOptions: {
-            values: false,
-            calcs: ['lastNotNull'],
-          },
-        },
-        fieldConfig: {
-          defaults: {
-            color: {
-              mode: FieldColorModeId.ContinuousGrYlRd,
-            },
-          },
-          overrides: [],
-        },
-      });
+export const barGaugeSugggestionsSupplier: VisualizationSuggestionsSupplierFn<Options> = (dataSummary) => {
+  if (!dataSummary.hasData || !dataSummary.hasFieldType(FieldType.number)) {
+    return;
+  }
 
-      list.append({
-        name: SuggestionName.BarGaugeLCD,
-        options: {
-          displayMode: BarGaugeDisplayMode.Lcd,
-          orientation: VizOrientation.Horizontal,
-          reduceOptions: {
-            values: false,
-            calcs: ['lastNotNull'],
-          },
-        },
-        fieldConfig: {
-          defaults: {
-            color: {
-              mode: FieldColorModeId.ContinuousGrYlRd,
-            },
-          },
-          overrides: [],
-        },
-      });
-    }
+  // This is probably not a good option for many numeric fields
+  if (dataSummary.fieldCountByType(FieldType.number) > BAR_LIMIT) {
+    return;
   }
-}
+
+  const suggestions: Array<VisualizationSuggestion<Options>> = [
+    { name: t('bargauge.suggestions.basic', 'Bar gauge') },
+    {
+      name: t('bargauge.suggestions.lcd', 'Bar gauge - LCD'),
+      options: {
+        displayMode: BarGaugeDisplayMode.Lcd,
+      },
+    },
+  ];
+
+  const shouldDeaggregate =
+    dataSummary.hasFieldType(FieldType.string) &&
+    dataSummary.frameCount === 1 &&
+    dataSummary.rowCountTotal <= BAR_LIMIT;
```

Comment:
deaggreagate is a strange term, the check has nothing to do with de-aggregation, but if we should display raw values or not